app/components/admin/participants/details.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Admin
+  module Participants
+    class Details < BaseComponent
+      def initialize(profile:)
+        @profile = profile
+        @variant = [profile.participant_type, (:pending if profile.pending?)].compact.join("_")
+      end
+
+    private
+
+      attr_reader :profile
+    end
+  end
+end
```

Comment:
Where can I see html for that? Is there some magic that means `@variant` just renders the right template?

Reply:
Yup, that's the magic with that weird file name. Honestly, I do not like it at all. I might issue a PR to ViewComponents to allow `component_name/#{variant}.html.erb` which feels so much better..